src/codegen/templates/library_interpreter/exec_cdecl_c_function_call.py.mako

```diff
@@ -0,0 +1,37 @@
+<%page args="function"/>\
+<%
+    from codegen.utilities.interpreter_helpers import generate_interpreter_function_call_args, get_instantiation_lines_for_varargs, get_varargs_parameters, get_argument_definition_lines_for_varargs
+    from codegen.utilities.function_helpers import get_arguments_type
+    from codegen.utilities.text_wrappers import wrap, docstring_wrap
+%>\
+        args = [device_name]
+        argtypes = [ctypes_byte_str]
+<%
+    varargs_parameters = get_varargs_parameters(function)
+    
+    ## This is under the assumption that the varargs passes are all arrays of the same size.
+    varargs_array_length = f"len({varargs_parameters[0].parameter_name})"
+    instantiation_lines = get_instantiation_lines_for_varargs(function)
+    argument_definition_lines = get_argument_definition_lines_for_varargs(varargs_parameters)
+%>
+        for index in range(${varargs_array_length}):
+    %if len(instantiation_lines) > 0:
+        %for instantiation_line in instantiation_lines:
+            ${instantiation_line}
+        %endfor
+    %endif
+
+    %if len(argument_definition_lines) > 0:
+        %for argument_definition_line in argument_definition_lines:
+            ${argument_definition_line}
+        %endfor
+    %endif
+        args.append(None)
```

Comment:
The [existing implementation](https://github.com/ni/nidaqmx-python/blob/f64f626aaf349fd44bddd7915c88be8ce1457f14/generated/nidaqmx/system/system.py#L274) of these types of methods does not define an argument for these, and hence I followed the same here. so is argtypes required for this case where we provide the None parameter as a means for varargs termination?.

Reply:
I see. We don't **really** need to add varargs to argtypes at all. We only need to add the non-variadic arguments: https://docs.python.org/3/library/ctypes.html#calling-variadic-functions

However, we can't just remove the varargs, because ctypes converts `str` to `wchar_t*` by default: https://docs.python.org/3/library/ctypes.html#calling-functions

> None, integers, bytes objects and (unicode) strings are the only native Python objects that can directly be used as parameters in these function calls. None is passed as a C NULL pointer, bytes objects and strings are passed as pointer to the memory block that contains their data (char* or wchar_t*). Python integers are passed as the platforms default C int type, their value is masked to fit into the C type.

In my opinion, if we are going to add the varargs to argtypes, we should add all of the varargs to argtypes.